music21/analysis/discrete.py

```diff
@@ -970,8 +975,8 @@ def _generateColors(self, numColors=None):
         if numColors is None:
             if self._referenceStream is not None:
                 # get total range for entire piece
-                minPitchObj, maxPitchObj = self.getPitchSpan(self._referenceStream)
-                difference = int(maxPitchObj.ps - minPitchObj.ps)
+                self.minPitchObj, self.maxPitchObj = self.getPitchSpan(self._referenceStream)
+                difference = int(self.maxPitchObj.ps - self.minPitchObj.ps)
                 minPitch, maxPitch = 0, difference
             else:
                 minPitch, maxPitch = 0, 130  # a large default
```

Comment:
I eventually want to get rid of this paradigm of setting two variables by creating two tuples that reference each other.

```
minPitch = 0
maxPitch = difference
```

seems so much clearer.  And in fact, it uncovers that `minPitch = 0` should be outside the `if` statement.